orchestrator/utils/errors.py

```diff
@@ -90,57 +104,29 @@ def is_api_exception(ex: Exception) -> bool:
     return ex.__class__.__name__ == "ApiException"
 
 
-def error_state_to_dict(err: ErrorState) -> ErrorDict:
-    """Return an ErrorDict based on the exception, string or tuple in the ErrorState.
+def error_state_to_dict(err: Exception) -> ErrorDict:
+    """Return an ErrorDict based on the exception.
 
     Args:
-        err: ErrorState from a workflow or api error state
-
+        err: Exception
     Returns:
         An ErrorDict containing the error message a status_code and a traceback if available
 
     """
-    # Import here to prevent cyclic imports
-    from orchestrator.forms import FormNotCompleteError, FormValidationError
-
-    if isinstance(err, FormValidationError):
+    if isinstance(err, ProcessFailure):
+        return {"class": type(err).__name__, "error": err.message, "traceback": show_ex(err), "details": err.details}
+    elif is_api_exception(err):
+        err = cast(ApiException, err)
         return {
             "class": type(err).__name__,
-            "error": str(err),
+            "error": err.reason,
```

Comment:
It seems like the API is changing? We don't need a schema change?

Reply:
No why would we? Btw the diff on this is not great, make sure to set "hide whitespace changes" to on

Reply:
Ah correct ;)

Reply:
The FormValidationError and FormNotCompleteError are not handled by this function (since fastapi) and we only use this function with actual exceptions instead of anything goes (Since removal of sn7) so I just simplified it a lot and added support for ProcessFailure errors